### ISSUE TITLE
fix(training): complete RNG seeding via set_global_seed in distributional RL and predictive planner scripts

### ISSUE THREAD

**Issue by ll7** [ll7/robot_sf_ll7#6499]
Parent issue: #6483

## Goal
Complete RNG seeding in the two training scripts that still use ad-hoc seeding, so every training script routes through the canonical `common.set_global_seed` (seeds `random`, `numpy`, `torch`, and CUDA determinism) defined in `robot_sf/common/seed.py`.

## Scope
- `scripts/training/train_distributional_rl.py:397-398`: replace `random.seed(config.seed)` + `torch.manual_seed(config.seed)` with `common.set_global_seed(config.seed)` (currently missing numpy).
- `scripts/training/train_predictive_planner.py:759-760`: replace `np.random.seed(args.seed)` + `torch.manual_seed(args.seed)` with `common.set_global_seed(args.seed)` (currently missing `random`).
- Add `from robot_sf import common` import where missing; remove now-unused `random`/`numpy` seed imports only if they become unused.
- Add a focused test proving `common.set_global_seed` is invoked with the configured seed in both scripts and that the prior ad-hoc calls are gone.

## Non-goals
- No change to `scripts/training/train_ppo.py` (config-key validation is child #6489; config hash is a separate child).
- No change to `robot_sf/common/seed.py` or anything under `robot_sf/benchmark/`.
- No training run or hyperparameter change.

## Validation
- Focused seeding tests for both scripts.
- `uv run ruff check` and `uv run ruff format --check` on changed files.
- `uv run pytest tests` passes (acceptance criterion).

## Factory v2 implementation contract
```json
{
  "allowed_paths": [
    "scripts/training/train_distributional_rl.py",
    "scripts/training/train_predictive_planner.py",
    "tests/training"
  ],
  "forbidden_paths": [
    "robot_sf/benchmark",
    "robot_sf/common/seed.py",
    "scripts/training/train_ppo.py"
  ],
  "objective": "Make scripts/training/train_distributional_rl.py and scripts/training/train_predictive_planner.py seed all of random, numpy, and torch plus CUDA determinism by calling common.set_global_seed with the configured seed, replacing the current partial ad-hoc seeding, and add regression tests proving the canonical helper is invoked with that seed.",
  "stop_conditions": [
    "Replace random.seed/torch.manual_seed at train_distributional_rl.py:397-398 with common.set_global_seed(config.seed), and np.random.seed/torch.manual_seed at train_predictive_planner.py:759-760 with common.set_global_seed(args.seed), so random, numpy, torch, and CUDA are all seeded through robot_sf/common/seed.py.",
    "Do not modify robot_sf/common/seed.py (read-only reference) or anything under robot_sf/benchmark/; the canonical helper already seeds all three RNGs plus CUDA.",
    "Do not modify scripts/training/train_ppo.py; config-key validation is owned by child #6489 and config hash by the sibling child.",
    "Add a focused test (e.g. monkeypatch common.set_global_seed) asserting it is called with the configured seed in both scripts and that the previous ad-hoc random.seed/np.random.seed/torch.manual_seed calls are removed.",
    "uv run pytest tests passes and ruff check/format are clean on the changed files."
  ],
  "validation": [
    "uv run ruff check scripts/training/train_distributional_rl.py scripts/training/train_predictive_planner.py",
    "uv run ruff format --check scripts/training/train_distributional_rl.py scripts/training/train_predictive_planner.py",
    "uv run pytest tests/training/test_train_distributional_rl_config.py tests/training/test_train_distributional_rl_dry_run.py tests/training/test_train_predictive_planner_dataset_diagnostics.py tests/test_seed_utils.py -q",
    "uv run pytest tests -q"
  ]
}
```
